x/bsonx/bsoncore/bsoncore_test.go

```diff
@@ -899,6 +900,52 @@ func TestBuild(t *testing.T) {
 	}
 }
 
+func TestNullBytes(t *testing.T) {
+	// Helper function to execute the provided callback and assert that it panics with the expected message. The
+	// createBSONFn callback should create a BSON document/array/value and return the stringified version.
+	assertBSONCreationPanics := func(t *testing.T, createBSONFn func(), expected string) {
+		t.Helper()
+
+		defer func() {
+			got := recover()
+			assert.Equal(t, expected, got, "expected panic with error %v, got error %v", expected, got)
+		}()
+		createBSONFn()
+	}
+
+	t.Run("element keys", func(t *testing.T) {
+		createDocFn := func() {
+			BuildDocument(nil, AppendStringElement(nil, "a\x00", "foo"))
```

Comment:
Thanks for pointing out the minor change from the commit on `master` here. Looks like `NewDocumentBuilder` is not available until 1.5.0 with GODRIVER-613.

LGTM even more, and no need to mark as 1.3.8 since we won't release on 1.3.